test/linalg/special.jl

```diff
@@ -1,59 +1,54 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 using Base.Test
-debug = false
 
 n= 10 #Size of matrix to test
 srand(1)
 
-debug && println("Test interconversion between special matrix types")
-let a=[1.0:n;]
-   A=Diagonal(a)
-   for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, Matrix]
-       debug && println("newtype is $(newtype)")
+@testset "Interconversion between special matrix types" begin
+   a = [1.0:n;]
+   A = Diagonal(a)
+   @testset for newtype in [Diagonal, Bidiagonal, SymTridiagonal, Tridiagonal, Matrix]
        @test full(convert(newtype, A)) == full(A)
    end
 
-   for isupper in (true, false)
-       debug && println("isupper is $(isupper)")
-       A=Bidiagonal(a, [1.0:n-1;], ifelse(isupper, :U, :L))
-       for newtype in [Bidiagonal, Tridiagonal, Matrix]
-           debug && println("newtype is $(newtype)")
+   @testset for isupper in (true, false)
+       A = Bidiagonal(a, [1.0:n-1;], ifelse(isupper, :U, :L))
+       @testset for newtype in [Bidiagonal, Tridiagonal, Matrix]
```

Comment:
same here: `@testset "conversion from Bidiagonal" begin`, or completely remove the set.